cmd/fyne/internal/commands/package_test.go

```diff
@@ -32,6 +32,10 @@ func Test_fixedVersionInfo(t *testing.T) {
 		{"1.1.1.1", goversioninfo.FileVersion{Major: 1, Minor: 1, Patch: 1, Build: 1}},
 		{"2.2.2", goversioninfo.FileVersion{Major: 2, Minor: 2, Patch: 2, Build: 1}},
 		{"3.3.3.3.3", goversioninfo.FileVersion{Major: 3, Minor: 3, Patch: 3, Build: 3}},
+		{"4.4.4-dev", goversioninfo.FileVersion{Major: 4, Minor: 4, Patch: 4, Build: 1}},
+		{"5.5.5+5", goversioninfo.FileVersion{Major: 5, Minor: 5, Patch: 5, Build: 1}},
+		{"6.6-foo+6", goversioninfo.FileVersion{Major: 6, Minor: 0, Patch: 0, Build: 1}},
+		{"7.7.7.7-foo", goversioninfo.FileVersion{Major: 7, Minor: 7, Patch: 7, Build: 1}},
```

Comment:
Why is this not a "-foo" modifier on 7.7.7.7?

Reply:
I think I solved this a lot more elegantly now, but I have no idea why some tests are failing - seems more like a problem with the pipelines?

Reply:
Tests all sorted - do the comments make sense now? I've tried to document the fact that the behaviour might be non-intuitive since no errors are thrown.
The code is is hopefully as DWIM as possible now and won't surprise anyone.